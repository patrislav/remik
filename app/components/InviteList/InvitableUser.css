.wrapper {
  border: none;
  padding: 0.3rem;
  background-color: inherit;
  width: 100%;
  display: flex;
  align-items: center;
  transition: all 250ms ease-in-out;
}

.wrapper:hover {
  background-color: #eee;
}

.photo {
  width: 36px;
  height: 36px;
  border-radius: 50%;
}

.name {
  margin-left: 0.3rem;
}
