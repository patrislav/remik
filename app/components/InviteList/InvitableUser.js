import React, {PropTypes} from 'react'
import styles from './InvitableUser.css'

const propTypes = {
  user: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    photoURL: PropTypes.string.isRequired
  }).isRequired
}

const InvitableUser = ({ user, ...props }) => (
  <button className={styles.wrapper} {...props}>
    <img src={user.photoURL} className={styles.photo} />
    <span className={styles.name}>{user.name}</span>
  </button>
)

InvitableUser.propTypes = propTypes

export default InvitableUser
