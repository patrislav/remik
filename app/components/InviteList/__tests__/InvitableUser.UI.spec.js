import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'

import InvitableUser from '../InvitableUser'

describe('<InvitableUser />', () => {
  it('renders correct UI', () => {
    const user = {
      id: 'ID',
      name: 'NAME',
      photoURL: 'PHOTO_URL'
    }
    const wrapper = shallow(<InvitableUser user={user} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
