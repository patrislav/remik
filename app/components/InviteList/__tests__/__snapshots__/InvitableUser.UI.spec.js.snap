exports[`<InvitableUser /> renders correct UI 1`] = `
<button
  className="wrapper">
  <img
    className="photo"
    src="PHOTO_URL" />
  <span
    className="name">
    NAME
  </span>
</button>
`;
